### ISSUE TITLE
BucketIterator splits function takes batch values seperately for train valid and test split

### ISSUE THREAD

**Issue by thak123** [bentrevett/pytorch-seq2seq#68]
> BucketIterator.splits(
>     (train_data, valid_data, test_data), 
>     batch_size = BATCH_SIZE, 
>     device = device)
> 

 _batch_size_ is replaced by _**batch_sizes**_ which like the first argument datasets takes a tuple of ints

BucketIterator.splits(
    (train_data, valid_data, test_data), 
    batch_sizes = (BATCH_SIZE,)*3, // either way (BATCH_SIZE,BATCH_SIZE,BATCH_SIZE)
    device = device)

**Comment by bentrevett** [comment]
I think as we are using a single batch size for train, valid and test data then we should stick to using `batch_size`. 

Could potentially use bigger batch sizes for valid/test (as we don't need to backprop) and if so, I would switch to `batch_sizes`. Will check what speed-ups we get and if it's considerable then I'll make the change.

Thanks for the suggestion.

**Comment by thak123** [comment]
I am sorry for the confusion. You are correct. Kindly ignore. Also it would be great if one notebook could be created explaining the gradient accumulation with Pytorch.

Thanks for such a wonderful tutorials
